src/PubSubClient.cpp

```diff
@@ -453,46 +453,15 @@ bool PubSubClient::publish_P(const char* topic, const char* payload, bool retain
 }
 
 bool PubSubClient::publish_P(const char* topic, const uint8_t* payload, size_t plength, bool retained) {
-    uint8_t llen = 0;
-    uint8_t digit;
-    size_t rc = 0;
-    size_t tlen;
-    size_t pos = 0;
-    const uint8_t header = MQTTPUBLISH | (retained ? MQTTRETAINED : 0);
-    size_t len;
-    size_t expectedLength;
-
-    if (!connected()) {
-        return false;
-    }
-
-    tlen = strnlen(topic, this->bufferSize);
-
-    this->buffer[pos++] = header;
-    len = plength + 2 + tlen;
-    do {
-        digit = len & 127;  // digit = len % 128
-        len >>= 7;          // len = len / 128
-        if (len > 0) {
-            digit |= 0x80;
+    if (beginPublish(topic, plength, retained)) {
```

Comment:
Shouldn't this also expect the topic to be in pgm?
So the `beginPublish` function should then use `strlen_P` just to be sure.

Reply:
True. ATM it is also not supported by the PubSubClient. I've seen somewhere a feature wish to have this. As this should only be a refactoring it should go to a new PR / issue ...